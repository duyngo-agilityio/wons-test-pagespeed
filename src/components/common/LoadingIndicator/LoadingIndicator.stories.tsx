// Libs
import type { Meta, StoryObj } from '@storybook/react';

// Components
import LoadingIndicator from './index';

const meta: Meta<typeof LoadingIndicator> = {
  title: 'Components/Common/LoadingIndicator',
  component: LoadingIndicator,
  args: {},
  argTypes: {
    size: {
      description: 'Change the size of the loading indicator',
      options: ['sm', 'md', 'lg'],
      control: { type: 'select' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof LoadingIndicator>;

export const Default: Story = {
  args: { size: 'lg' },
  render: () => (
    <div className="w-96 h-48">
      <LoadingIndicator />
    </div>
  ),
};

export const Small: Story = {
  args: { size: 'sm' },
  render: () => (
    <div className="w-96 h-48">
      <LoadingIndicator size="sm" />
    </div>
  ),
};

export const Medium: Story = {
  args: { size: 'md' },
  render: () => (
    <div className="w-96 h-48">
      <LoadingIndicator size="md" />
    </div>
  ),
};

export const Large: Story = {
  args: { size: 'lg' },
  render: () => (
    <div className="w-96 h-48">
      <LoadingIndicator size="lg" />
    </div>
  ),
};
