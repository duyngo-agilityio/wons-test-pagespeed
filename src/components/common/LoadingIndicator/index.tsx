// Libs
import { memo } from 'react';

// Components
import { Spinner } from '@/components/common';

interface ILoadingIndicatorProps {
  size?: 'sm' | 'md' | 'lg';
}

const LoadingIndicator = ({ size = 'lg' }: ILoadingIndicatorProps) => (
  <div className="fixed inset-0 flex items-center justify-center bg-black bg-opacity-70 z-50">
    <Spinner size={size} />
  </div>
);

export default memo(LoadingIndicator);
