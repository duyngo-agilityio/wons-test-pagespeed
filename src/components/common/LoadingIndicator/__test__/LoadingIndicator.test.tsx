// Components
import LoadingIndicator from '../index';

describe('LoadingIndicator component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match with snapshot', () => {
    const { container } = testLibJestUtils.render(<LoadingIndicator />);

    expect(container).toMatchSnapshot();
  });
});
