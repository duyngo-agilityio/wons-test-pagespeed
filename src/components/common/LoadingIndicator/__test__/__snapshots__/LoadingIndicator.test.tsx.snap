// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component should match with snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
    <div
      class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-70 z-9999"
    >
      <div
        aria-label="Loading"
        class="relative inline-flex flex-col gap-2 items-center justify-center"
      >
        <div
          class="relative flex w-10 h-10"
        >
          <i
            class="absolute w-full h-full rounded-full animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent border-3 border-b-blue-500"
          />
          <i
            class="absolute w-full h-full rounded-full opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent border-3 border-b-blue-500"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
